src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -362,10 +362,6 @@ protected virtual IEnumerable<SyntaxNode> DefaultNodesExtractor(SyntaxNode node,
                 foreach (var headerNode in ExtractNodesOfHeader(node, syntaxFacts))
                 {
                     yield return headerNode;
-                    foreach (var extractedNode in ExtractNodesSimple(headerNode, syntaxFacts))
-                    {
-                        yield return extractedNode;
-                    }
                 }
             }
         }
```

Comment:
totally a nit, but you could also consider writing this as:

```c#
var simpleNodes = ExtractNodesSimple(node, syntaxFacts);
var headerNodes = extractParentsOfHeader ? ExtractNodesOfHeader(node, syntaxFacts) : SpecializedCollections.Empty<SyntaxNode>();
return simpleNodes.Concat(headerNodes);
```

Reply:
Also: in general we take a wary position toward IEnumerable (though no need to change here, just be careful).  The primary concern is that it is very easy for the IEnumerable to be both lazy and expensive.  And just simple usage patterns can cause perf problems down the line.  for example, if higher up code did `.Any()` on it, and then did some other operation, this would could cause two full walks of the entire stream.  

I don't think there's any real concern here. But it's something to watch out for as it's bitten us a lot in the past (and it's one of the reason i'm rewriting SymbolKey).  Using a pooled builder that the values can be written into is nice and simple and clear in all levels as to the overall CPU and memory costs involved.

Because this is part of the refactoring codepath, i'm not concerned.  But if this were part of a lower level area that might be used in a hotspot, i'd be a little more concerned.

Reply:
This will get substantially changed in next PR. The helper will directly call either ExtractIfInHeader and ExtractSimple (should come up with a better name) because I need to change ExtractIfInHeader to operate on Tokens instead.

As of being cautions with IEnumerable, I know.

Reply:
Sounds good :)